File: src/main/java/tn/itbs/prod/services/ProduitService.java
package tn.itbs.prod.services;

import java.util.List;

import tn.itbs.prod.models.Produit;

public interface ProduitService {
	List<Produit> getAllProduits();
    Produit getProduitById(Long id);
    Produit createProduit(Produit produit);
    Produit updateProduit(Long id, Produit produit);
    void deleteProduit(Long id);

}

File: frontend/gestion-production-ui/src/app/services/produit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ProduitService {
  private baseUrl = 'http://localhost:8080/api/produits'; 

  constructor(private http: HttpClient) {}

  getAllProduits(): Observable<any> {
    return this.http.get(this.baseUrl);
  }

  createProduit(produit: any): Observable<any> {
    return this.http.post(this.baseUrl, produit);
  }

  getProduitById(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  updateProduit(id: number, produit: any): Observable<any> {
    return this.http.put(`${this.baseUrl}/${id}`, produit);
  }

  deleteProduit(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`);
  }

  
}